experimental/benchmarks/node-dc-eis/run_node-dc-eis.sh

```diff
@@ -0,0 +1,297 @@
+    usage
+    exit
+  fi
+}
+
+function optional() {
+  if [ -z "${!1}" ]; then
+    echo -n "${1} not set (ok)"
+    if [ -n "${2}" ]; then
+      echo -n ", default is: ${2}"
+      export ${1}="${2}"
+    fi
+    echo ""
+  fi
+}
+
+function remove(){
+  if [ -f $1 ]; then
+    rm $1
+  fi
+}
+
+function archive_files() {
+  # archive files
+  echo -e "\n##BEGIN $TEST_NAME Archiving $(date)\n"
+  mv $LOGDIR_TEMP/$LOGDIR_PREFIX $RESULTSDIR
+  echo -e "Perf logs stored in $RESULTSDIR/$LOGDIR_PREFIX"
+  echo -e "\nCleaning up"
+  rm -r $LOGDIR_TEMP
+  echo -e "\n## END $TEST_NAME Archiving $(date)\n"
+}
+
+function kill_bkg_processes()   # kill processes started in background
+{
+  echo "Killing due to : $@"    
+  $MONGODB_COMMAND "stop"
+  pkill node
+  pids=$(ps -eo pid,pgid | awk -v pid=$$ '$2==pid && $1!=pid {print $1}')  # get list of all child/grandchild pids - this doesnt seem to work on nodejs benchmark machine....
+  echo "Killing background processes"
+  echo $pids
+  kill -9 $OTHERPID_LIST $pids || true  # avoid failing if there is nothing to kill
+}
+
+function on_exit()
+{
+  echo "Caught kill"
+  kill_bkg_processes "caught kill"
+  #kill -9 $PID_LIST $OTHERPID_LIST
+  archive_files
+  exit 1
+}
+
+function timestamp()
+{
+  date +"%Y%m%d-%H%M%S"
+}
+
+trap on_exit SIGINT SIGQUIT SIGTERM
+
+# VARIABLE SECTION
+
+# define variables
+declare -rx SCRIPT=${0##*/}
+TEST_NAME=node-dc-eis
+echo -e "\n## TEST: $TEST_NAME ##\n"
+echo -e "## OPTIONS ##\n"
+
+optional RESULTSDIR ${ROOT_DIR}/results
+optional TIMEOUT 600
+RESULTSLOG=$TEST_NAME.log
+SUMLOG=score_summary.txt
+
+optional DRIVERHOST
+optional NODE_FILE server-cluster
+optional CLUSTER_MODE false
+optional PORT 9000
+optional DRIVERCMD ${RESOURCE_DIR}/Node-DC-EIS-client/runspec.py
+optional DRIVERNO 25
+NODEDC_DRIVER_PATH=${SCRIPT_DIR}/Node-DC-EIS-client
+
+NODE_SERVER=$(hostname -s)
+                    
+
+echo -e "RESULTSDIR: $RESULTSDIR"
+echo -e "RESULTSLOG: $RESULTSLOG"
+echo -e "TIMEOUT: $TIMEOUT"
+echo -e "NODE_SERVER: $NODE_SERVER"
+echo -e "PORT: $PORT"
+echo -e "NETWORKTYPE: $NETWORKTYPE"
+echo -e "DRIVERCMD: $DRIVERCMD"
+echo -e "DRIVERNO: $DRIVERNO\n"
+echo -e "MONGODB_URL: $DB_URL\n"
+
+RUNSPEC_LOGFILE=$SCRIPT_DIR/nodedc-runspec.log
+DRIVER_COMMAND="$RUNSPEC_AFFINITY python $DRIVERCMD -f ${RESOURCE_DIR}/Node-DC-EIS-client/config.json -dir ${RESULTSDIR}"
+# END VARIABLE SECTION
+
+# Date stamp for result files generated by this run
+CUR_DATE=$(timestamp)
+
+PLATFORM=`/bin/uname | cut -f1 -d_`
+echo -e "Platform identified as: ${PLATFORM}\n"
+
+case ${PLATFORM} in
+  Linux)
+    bash ${SCRIPT_DIR}/kill_node_linux
+    ;;
+esac
+if [ -z $NODE ]; then
+  NODE=`which node`
+fi
+if [ -z "$NODE" ]; then
+  echo "ERROR: Could not find a 'node' executable. Please set the NODE environment variable or update the PATH."
+  echo "node is not here: $NODE"
+  exit 1
+fi
+
+echo -e "NODE VERSION:"
+$NODE --version
+# build command
+CMD="$NODE_AFFINITY ${NODE} ${NODE_FILE}" 
```

Comment:
I see that you use cache to save employee information rather than going to mongodb. Is there any reason why you set a cap on maxsize/maxAge [here](https://github.com/Node-DC/Node-DC-EIS/blob/master/Node-DC-EIS-cluster/config/configuration.js#L30-L31)? Do you see memory growth if you set maxSize to default i.e. Infinite? I think the goal should be to use as much cache as you can rather than relying on mongodb because that adds other factors to the benchmark score.